src/mixins/maskable.js

```diff
@@ -61,30 +61,22 @@ export default {
     // caret location is correct
     setSelectionRange () {
       this.$nextTick(() => {
-        if (this.$refs.input == null) return
+        if (!this.$refs.input) return
 
-        const maskedText = this.maskText(this.lazyValue)
-        if (maskedText !== this.$refs.input.value) {
-          this.$refs.input.value = maskedText
-        }
-
-        // May not be needed
-        // if (this.oldValue.length === this.selection) return
+        this.$refs.input.value = this.maskText(this.lazyValue)
 
         const newValue = this.$refs.input.value
         if (!this.deleting) {
-          while (isMaskDelimiter(newValue.slice(this.selection - 1, this.selection))) {
+          while (isMaskDelimiter(newValue.substr(this.selection - 1, 1))) {
             this.selection++
           }
+          // TODO: This only helps sometimes
+          if (this.$refs.input.selectionStart - this.selection === 1) {
+            this.selection = this.$refs.input.selectionStart
```

Comment:
How does it help?

Please try this:-

1. Type 035
2. Move the caret anywhere in between 3 and 5
3. Type 4
4. The caret moves right after 5 due to this added lines, while it should be in between 4 and 5

Using #1896 and:
```javascript
computed: {
  mask () { // There are three common formats for AU phone numbers
    return this.phone.length > 1 && this.phone.startsWith('0')
      ? this.phone.startsWith('04')
        ? '#### ### ###' // Mobile
        : '(##) #### ####' // With area code
      : '#### ####' // Local number
  }
}
```

Reply:
Comment it out and you get this instead ![image](https://user-images.githubusercontent.com/16421948/30846335-8bf81f78-a2da-11e7-97cc-efc6ab7c8f8d.png)

It only checks for a difference of 1, so if more than one character changes then the cursor goes to the end.

Reply:
I just realised that it breaks `fill-mask-blanks` -_-